### PR TITLE
Fixes dusting.

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

This fixes a lot of stuff about dusting.

- It now animates properly, you will no longer drop to the floor while dusting.
- Fixes an oversight where all the items are deleted before the dust animation plays, which causes modsuits to act funky on the dust animation.
- When a carbon is destroyed, along with all the items they are wearing, they are now silently unequipped. You won't hear a jumpsuit dropped sound if you delete an assistant for example. (This is with the exception of modsuits, because there is no suitable way to silence them currently right now. It would require either some hackery, or a refactor.)
- Codewise, this changes around `get_all_slots` so that the backslot is first, so that all the places that call it will always call the modsuit first if needed. I checked all the other cases, it'll fix some other hidden issues too where the modsuit would fuck up on deletion.

## Why It's Good For The Game
Bug fixes, animation looks good again

## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/91113370/06212269-0edd-486b-89b7-a8b508cdd1f8

https://github.com/ParadiseSS13/Paradise/assets/91113370/b980a4bf-1c6d-49dc-a29e-266079be8288

## Testing

Dusted myself
dusted myself in a modsuit
could still crawl around
couldnt lie down with the FORCED_STANDING trait

## Changelog
:cl:
fix: Dusting animation works properly again
fix: You will no longer hear the unequip sounds when someone is deleted.
/:cl:

### CODE REVIEW

File: code/_globalvars/traits.dm [hunk 1]
```diff
@@ -82,7 +82,8 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_TABLE_LEAP" = TRAIT_TABLE_LEAP,
 		"TRAIT_DODGE_ALL_THROWN_OBJECTS" = TRAIT_DODGE_ALL_OBJECTS,
 		"TRAIT_SUPERMATTER_IMMUNE" = TRAIT_SUPERMATTER_IMMUNE,
-		"TRAIT_BADASS" = TRAIT_BADASS
+		"TRAIT_BADASS" = TRAIT_BADASS,
+		"TRAIT_FORCED_STANDING" = TRAIT_FORCED_STANDING,
 	),
```
Comment (Henri215) [inline code/_globalvars/traits.dm:87]:
```suggestion
		"TRAIT_FORCED_STANDING" = TRAIT_FORCED_STANDING
	),
```

File: code/modules/mod/modules/modules_supply.dm [hunk 2]
```diff
@@ -428,6 +428,8 @@
 	mod.wearer.add_filter("mod_outline", 3, outline_filter(color = "#000000AA"))
 	animate(mod.wearer, animate_time, pixel_y = mod.wearer.pixel_y - 4, flags = ANIMATION_PARALLEL)
 	mod.wearer.SpinAnimation(1.5)
+	// todo, someone get balance approval to add TRAIT_FORCED_STANDING here, like it is on tg.
```
Comment (Henri215) [inline code/modules/mod/modules/modules_supply.dm:431]: Someone 🤣

Comment (Contrabang) [reply code/modules/mod/modules/modules_supply.dm:431]: out of scope :), but it would fix things

Comment (lewcc) [reply code/modules/mod/modules/modules_supply.dm:431]: Just get a PR up to handle it after this one goes in

**Review by lewcc — APPROVED** [review]
Code looks good

**Pull request merged**